msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/ADAP3DFeatureDetectionMethod.java

```diff
@@ -19,22 +19,31 @@
 
 import org.apache.commons.lang3.ArrayUtils;
 
+import io.github.msdk.MSDKMethod;
 import io.github.msdk.datamodel.impl.SimpleChromatogram;
 import io.github.msdk.datamodel.impl.SimpleFeature;
+import io.github.msdk.featdet.ADAP3D.common.algorithms.CurveTool;
+import io.github.msdk.featdet.ADAP3D.common.algorithms.Parameters;
+import io.github.msdk.featdet.ADAP3D.common.algorithms.PeakDetection;
+import io.github.msdk.featdet.ADAP3D.common.algorithms.SliceSparseMatrix;
 
 /**
  * <p>
  * This class is used to run the whole ADAP3D algorithm and get peaks.
  * </p>
  *
  */
-public class ADAP3DAlgorithm {
+public class ADAP3DFeatureDetectionMethod implements MSDKMethod<List<SimpleFeature>>{
```

Comment:
Should be 
````public class ADAP3DFeatureDetectionMethod implements MSDKMethod<List<Feature>>{````
You can just cast the list when you return it. SimpleFeature is Feature, so it will work.